### PR TITLE
Adding support for ubuntu 15.04 and 16.04

### PR DESCRIPTION
**Pull request by mskurski**
There were few breaking changes in Ubuntu 15.04 and 16.04 which made this cookbook incompatible like:
- different provider (they went back to systemd)
- mongod service for ubuntu 16.04 needs some additional configuration

I got some problems with running Ubuntu 16.04 on Vagrant machine (private network required by Vagrant doesn't work properly on official vagrant box for Ubuntu 16.04 and network interfaces names were also changed and Vagrant expects eth0/eth1) so just to simplify this process you can use below configuration:

**Vagrantfile**

```
# this line should be commented or removed from your Vagrantfile
#config.vm.host_name = 'your host'

config.vm.box = "geerlingguy/ubuntu1604"
config.vm.provision :shell, path: "./bin/ubuntu_vm_fix.sh"
```

**./bin/ubuntu_vm_fix.sh**

```
#!/usr/bin/env bash

# https://github.com/mitchellh/vagrant/issues/7288
if ! grep -q "127.0.0.1 vagrant.local ubuntu-xenial" /etc/hosts;
then
  echo "#### adding hostname for private network to /etc/hosts (required by vagrant)"
  echo "127.0.0.1 vagrant.local ubuntu-xenial" | sudo tee -a /etc/hosts
fi

# fix network ubuntu/xenial
# disable Predictable Network Interface Names
ip a show dev eth1 > /dev/null
    if (($? > 0)); then
    echo "#### eth1 not exists (eth1 and eth2 is expected by vagrant)"
    echo "#### turning off Predictable Network Interface Names"
    grep -q "net.ifnames=0" /etc/default/grub
    if (($? > 0)); then
      echo "GRUB_CMDLINE_LINUX=\"net.ifnames=0"\" | sudo tee -a /etc/default/grub
      sudo update-grub
      echo "#### restart box to disable Predictable Network Interface Names"
    fi
fi
```

Just let me know if you will have any problems or questions

### CODE REVIEW

File: metadata.rb [hunk 1]
```diff
@@ -4,9 +4,9 @@
 license          'Apache 2.0'
 description      'Installs/Configures mongodb3'
 long_description 'Installs/Configures mongodb3'
-version          '5.2.0'
+version          '5.3.0'
 
-supports 'ubuntu', '= 12.04'
+supports 'ubuntu', '= 16.04'
```
Comment (goatherd) [inline metadata.rb]: Should be `supports 'ubuntu', '>= 12.04'` or unchanged.

**Comment by goatherd** [comment]
Works for me on a generic `"ubuntu/xenial64"` box.

File: recipes/package_repo.rb [hunk 2]
```diff
@@ -97,7 +97,11 @@
   when 'debian'
     apt_repository "mongodb-org-#{pkg_major_version}" do
       uri node['mongodb3']['package']['repo']['url']
-      distribution "#{node['lsb']['codename']}/mongodb-org/#{node['mongodb3']['package']['repo']['apt']['name']}"
+      if node['platform'] == 'ubuntu' and node['platform_version'].to_f >= 15.04
+        distribution "trusty/mongodb-org/#{node['mongodb3']['package']['repo']['apt']['name']}"
```
Comment (goatherd) [inline recipes/package_repo.rb:102]: http://repo.mongodb.org/apt/ubuntu/dists/xenial/mongodb-org/ exists for 16.04 so this should be `= 15.04` instead.

Looks like mongodb does only support LTS-Versions with their repository. Isn't support for `15.04` mostly a hack?

File: metadata.rb (hunk 1 above)
Comment (mskurski) [inline metadata.rb]: right, I will fix this

File: recipes/package_repo.rb (hunk 2 above)
Comment (mskurski) [inline recipes/package_repo.rb:102]: Right, seems so they support only LTS versions:
[Platform Support](https://docs.mongodb.com/manual/tutorial/install-mongodb-on-ubuntu/)

> MongoDB only provides packages for 64-bit long-term support Ubuntu releases. Currently, this means 12.04 LTS (Precise Pangolin) and 14.04 LTS (Trusty Tahr). While the packages may work with other Ubuntu releases, this is not a supported configuration.

I agree that support of 15.04 might be considered as a hack, but mongodb release for 14.04 works on 15.04 as well. I will fix this condition and force trusty/mongodb-org release only for 15.04

**Comment by mskurski** [comment]
Thank you @goatherd for your comments, I've already applied fixes.

**Comment by goatherd** [comment]
@mskurski 

Cool, that was fast.

Really helped me a lot!

**Comment by aight8** [comment]
?

**Comment by aight8** [comment]
Why this PR is open since 2 month? Come on guys. Can you decide a new maintainer when you have no time for this project please?

**Comment by sunggun-yu** [comment]
@aight8 sorry for the late. I was away from chef for several month. sorry again. I need some collaborators. lol
I'll update shortly.

**Comment by sunggun-yu** [comment]
@mskurski 
Thank you for the great feature. and sorry for late. 
it's merged and I'll release it once I finish some testing.

and guys, please forgive me :)
@aight8 @goatherd

**Comment by mskurski** [comment]
@sunggun-yu 
no worries mate :)

**Pull request merged**